auth.go

```diff
@@ -87,78 +97,96 @@ func DecodeBase64Auth(auth AuthConfig) (string, string, error) {
 var (
 	ErrCredentialsNotFound         = errors.New("credentials not found in native keychain")
 	ErrCredentialsMissingServerURL = errors.New("no credentials server URL")
 )
 
+//nolint:gochecknoglobals // These are used to mock exec in tests.
+var (
+	// execLookPath is a variable that can be used to mock exec.LookPath in tests.
+	execLookPath = exec.LookPath
+	// execCommand is a variable that can be used to mock exec.Command in tests.
+	execCommand = exec.Command
+)
+
 // GetCredentialsFromHelper attempts to lookup credentials from the passed in docker credential helper.
 //
-// The credential helpoer should just be the suffix name (no "docker-credential-").
+// The credential helper should just be the suffix name (no "docker-credential-").
 // If the passed in helper program is empty this will look up the default helper for the platform.
 //
 // If the credentials are not found, no error is returned, only empty credentials.
 //
-// Hostnames should already be resolved using `ResolveRegistryAuth`
+// Hostnames should already be resolved using [ResolveRegistryHost]
 //
 // If the username string is empty, the password string is an identity token.
 func GetCredentialsFromHelper(helper, hostname string) (string, string, error) {
 	if helper == "" {
-		helper = getCredentialHelper()
-	}
-	if helper == "" {
-		return "", "", nil
+		helper, helperErr := getCredentialHelper()
+		if helperErr != nil {
+			return "", "", fmt.Errorf("get credential helper: %w", helperErr)
+		}
+
+		if helper == "" {
+			return "", "", nil
+		}
 	}
 
-	p, err := exec.LookPath("docker-credential-" + helper)
+	helper = "docker-credential-" + helper
+	p, err := execLookPath(helper)
 	if err != nil {
+		if !errors.Is(err, exec.ErrNotFound) {
+			return "", "", fmt.Errorf("look up %q: %w", helper, err)
+		}
+
 		return "", "", nil
 	}
 
-	cmd := exec.Command(p, "get")
+	var outBuf, errBuf bytes.Buffer
+	cmd := execCommand(p, "get")
 	cmd.Stdin = strings.NewReader(hostname)
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
 
-	b, err := cmd.Output()
-	if err != nil {
-		s := strings.TrimSpace(string(b))
-
-		switch s {
+	if err = cmd.Run(); err != nil {
+		out := strings.TrimSpace(outBuf.String())
+		switch out {
 		case ErrCredentialsNotFound.Error():
 			return "", "", nil
 		case ErrCredentialsMissingServerURL.Error():
-			return "", "", errors.New(s)
+			return "", "", ErrCredentialsMissingServerURL
 		default:
+			return "", "", fmt.Errorf("execute %q stdout: %q stderr: %q: %w",
+				helper, out, strings.TrimSpace(errBuf.String()), err,
+			)
 		}
-
-		return "", "", err
 	}
 
 	var creds struct {
-		Username string
-		Secret   string
+		Username string `json:"Username"`
+		Secret   string `json:"Secret"`
 	}
 
-	if err := json.Unmarshal(b, &creds); err != nil {
-		return "", "", err
+	if err = json.Unmarshal(outBuf.Bytes(), &creds); err != nil {
+		return "", "", fmt.Errorf("unmarshal from: %q output: %q: %w", helper, outBuf.String(), err)
```

Comment:
I don't think this should include the output buffer data.

Reply:
Have removed this, but it will make it harder to debug any issue, but that should be rare so likely a good compromise :)